File: code/widgets/org.eclipse.scout.widgets.client/src/main/java/org/eclipse/scout/widgets/client/ui/forms/SimpleTileFilter.java
package org.eclipse.scout.widgets.client.ui.forms;

import org.eclipse.scout.rt.client.ui.tile.ITileFilter;
import org.eclipse.scout.rt.platform.util.ObjectUtility;
import org.eclipse.scout.widgets.client.ui.tile.AbstractSimpleTile;

public class SimpleTileFilter implements ITileFilter<AbstractSimpleTile> {
  private String m_text;

  public SimpleTileFilter() {
    setText("");
  }

  public void setText(String text) {
    text = ObjectUtility.nvl(text, "");
    m_text = text.toLowerCase();
  }

  @Override
  public boolean accept(AbstractSimpleTile tile) {
    String label = ObjectUtility.nvl(tile.getLabel(), "");
    String filterText = label.trim().toLowerCase();
    return filterText.indexOf(m_text) > -1;
  }

}

File: code/widgets/org.eclipse.scout.jswidgets.ui.html/src/main/js/jswidgets/tilegrid/SimpleTileFilter.js
jswidgets.SimpleTileFilter = function(model) {
  model = model || {};
  this.text = null;
  this.setText(model.text);
};

jswidgets.SimpleTileFilter.prototype.setText = function(text) {
  this.text = text || '';
  this.text = this.text.toLowerCase();
};

jswidgets.SimpleTileFilter.prototype.accept = function(tile) {
  var label = tile.label || '';
  var filterText = label.trim().toLowerCase();
  return filterText.indexOf(this.text) > -1;
};
